src/main/java/com/gotocompany/depot/message/ProtoUnknownFieldValidationType.java

```diff
@@ -0,0 +1,83 @@
+package com.gotocompany.depot.message;
+
+import com.google.protobuf.Message;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public enum ProtoUnknownFieldValidationType {
+    MESSAGE {
+        @Override
+        public boolean shouldFilter(Object object) {
+            return object instanceof Message;
+        }
+
+        @Override
+        public Stream<Message> getMapper(Object object) {
+            return Stream.of((Message) object);
+        }
+    },
+    MESSAGE_ARRAY_FIRST_INDEX {
+        @Override
+        public boolean shouldFilter(Object object) {
+            return isMessageOrMessageListType(object);
+        }
+
+        @Override
+        public Stream<Message> getMapper(Object object) {
+            if (object instanceof Message) {
+                return Stream.of((Message) object);
+            }
+            if (object instanceof List) {
+                return Optional.ofNullable(((List) object).get(0))
+                        .filter(this::shouldFilter)
+                        .map(o -> Stream.of((Message) o))
+                        .orElseGet(Stream::empty);
+            }
+            return Stream.empty();
+        }
+    },
+    MESSAGE_ARRAY_FULL {
+        @Override
+        public boolean shouldFilter(Object object) {
+            return isMessageOrMessageListType(object);
+        }
+
+        @Override
+        public Stream<Message> getMapper(Object object) {
+            if (object instanceof Message) {
+                return Stream.of((Message) object);
+            }
+            if (object instanceof List) {
+                return Optional.of(object)
+                        .filter(this::shouldFilter)
+                        .map(messageList -> ((List<Message>) messageList).stream())
+                        .orElseGet(Stream::empty);
```

Comment:
MESSAGE_ARRAY_FULL -> is aimed to only check whether it's a singular message or all of the repeated message.

The supplied shouldFilter should safely filter and make sure the processed element to be Message or List of message, wdyt?

Purpose of this change is because in the previous implementation, repeated Message is not being considered at all for checking whether there are unknown fields or not.